astropy/table/tests/test_column.py

```diff
@@ -156,6 +156,18 @@ def test_quantity_init(self, Column):
         assert np.all(c.data == np.array([100, 200, 300]))
         assert np.all(c.unit == u.cm)
 
+    def test_quantity_with_info_init(self, Column):
+        q = np.arange(3.0) * u.m
+        q.info.name = "q"
+        q.info.description = "an example"
+        q.info.meta = {"parrot": "dead"}
+        q.info.format = "3.1f"
+        c = Column(q)
+        assert c.name == "q"
+        assert c.description == "an example"
+        assert c.meta == q.info.meta and c.meta is not q.info.meta
```

Comment:
If this check ever fails then the traceback will be needlessly complicated.
```suggestion
        assert c.meta == q.info.meta
        assert c.meta is not q.info.meta
```

Reply:
Done (as part of edited commit 1)